lib/system/database.rb

```diff
@@ -148,22 +148,32 @@ def adapter_module
 end
 
 ActiveSupport.on_load(:active_record) do
-if System::Database.oracle? && defined?(ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks)
-  ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks.class_eval do
-    prepend(Module.new do
-
-      def create
-        super
-        connection.execute "GRANT create trigger TO #{username}"
-        connection.execute "GRANT create procedure TO #{username}"
-      end
+  if System::Database.oracle? && defined?(ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks)
+    ActiveRecord::ConnectionAdapters::OracleEnhancedAdapter::DatabaseTasks.class_eval do
+      prepend(Module.new do
+        # If ORACLE_SYSTEM_PASSWORD is provided, Porta impersonates Oracle's SYSTEM user to create/update a non-SYSTEM
+        # user and grants it permissions. This is a behaviour that should preferably be used during development as
+        # the usage of Oracle's SYSTEM user grants the application more power than it should have in the first place.
+        # Alternatively, Porta should be provided with a non-SYSTEM user that has been granted the necessary
+        # permissions.
+        def create
+          if ENV['ORACLE_SYSTEM_PASSWORD'].present?
+            Rails.logger.warn("Oracle's SYSTEM user will create/update a non-SYSTEM user and grant it permissions")
```

Comment:
In my local environment, this fails because `Rails.logger` is `nil`. Could you please double check this? Not sure if it's a problem on my side.

Reply:
Odd... @jlledom  Did you get the error while calling `rails db:create`?
Also, in which ENV did you try to run this code?

Reply:
> I don't see things failing for me but I don't see the console message either.

This could be because of the log level... Maybe we should set it for something else 🤔